Simulation section

dominikmuller opened the pull request:
Hi all,

Greg and I have finished some sort of simulation section which covers general MC related information + DecFiles + MC filtering for tomorrow's lecture.

As this is the first time we present something simulation related in the starter/impact kit I am sure we will have lots of ideas how to improve the material after the session tomorrow.

dominikmuller committed:
Added Greg's stuff

dominikmuller committed:
Migrated to lb-dev

dominikmuller committed:
Structure

dominikmuller committed:
Simulation figure 1

dominikmuller committed:
Added basic instructions how to find the correct option files

dominikmuller committed:
Introduction

dominikmuller committed:
Added filtering section

dominikmuller committed:
Merged Greg's part

dominikmuller committed:
Updated headings

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+1. The `ProductionTool` (Pythia, GenXicc, ...) generates an event with the required signal particle. Either by generating minimum bias events until a matching particle is found or by ensuring one is produced in every event.
+2. The signal particle is decayed using the `DecayTool` (EvtGen) to the desired final state, all remaining unstable particles are decayed independently.
+3. The signal and its decay products might be required to pass generator level cuts implemented as a `CutTool`.
+4. Particles are transported through the detector simulation.
```
I found the description in [LHCBGAUSS-1090](https://its.cern.ch/jira/browse/LHCBGAUSS-1090) to be quite interesting, because it describes the logic that decides which particles are decayed by EvtGen and which are decayed by Geant4.

> * if a particle is known to EvtGen, then it is declared stable in the production generator – it will be decayed by EvtGen
> * if a particle is not known to EvtGen, then it is decayed in the production generator
> * if a particle is not known to EvtGen or the production generator, it will be decayed by Geant4

Maybe it would be interesting to include something like this?

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+
+> ## Things to remember {.callout}
+>
+> 1. The detector simulation is the **__by far__** most time consuming step (minutes). So make sure your generator cuts remove events you cannot possible reconstruct or select later on. Additional options are available to increase the speed, please talk to your MC liaisons!
```
* "is the by far" → "is by far the"
* "(minutes)" → "(minutes, compared to seconds for the generation)"
* "possible reconstruct" → "possibly reconstruct"

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+> ## Things to remember {.callout}
+>
+> 1. The detector simulation is the **__by far__** most time consuming step (minutes). So make sure your generator cuts remove events you cannot possible reconstruct or select later on. Additional options are available to increase the speed, please talk to your MC liaisons!
+> 2. The generator cuts are only applied to the signal that was forced to decay to the specific final state. _Any_ other true candidate is not required to pass.
```
Maybe "Any other true signal candidate"?

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+First, find the ProductionID:
+![FindingProductionID](img/simulation_1.png)
+Search for this ID in the Transformation Monitor, right click the result and select "Show request". Right clicking and selecting "View" in the new window will open an overview about all the individual steps of the production with their application version and option files used.
+> Important: the order of the option files does matter!
```
I think you need to add something extra to here to use the special callout block style.

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+## Setting up a new Decay
+
+EvtGen is completely controlled via a specific file for each sample, known as a DecFile. These live in the DecFiles package:
+https://gitlab.cern.ch/LHCb-SVN-mirrors/Gen-DecFiles/tree/master/dkfiles
```
I'd make "DecFiles package" a hyperlink.

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+To understand what is produced in any simulated sample, you need to understand these. First, how to try them out.
+
+The procedure for testing and committing decfiles is well documented:
+https://twiki.cern.ch/twiki/bin/view/LHCb/GaussDecayFiles
```
Hyperlink "well documented" with the link.

alexpearce commented on simulation.md:
````diff
@@ -0,0 +1,454 @@
+```
+which should point `Gen/DecFiles` directory in the current development environment.
+
+>Note that recompiling will not overwrite existing options file, it is necessary to remove by hand all of the python files in `./Gen/DecFiles/Options`.
````
Because there's no newline after this quote, the next sentence will be in a quote as well. I don't think that's what you want.

alexpearce commented on simulation.md:
````diff
@@ -0,0 +1,454 @@
+```shell
+lb-run DaVinci/v41r0 gaudirun.py DaVinciOptions.py
+```
+This script will attempt to build an nTuple from the xgen file it is given, using the specified decay descriptor. If everything is working correctly, this should return one entry per event, corresponding to your signal candidate. The PrintMCTree algorithm will print to screen the full decay chain for each particle in "decay_heads" e.g:
````
I guess there _could_ be more than one entry per event.

alexpearce commented on simulation.md:
````diff
@@ -0,0 +1,454 @@
+#
+```
+The information in the header is not just bookkeeping, almost all of it is parsed and changes what you get out at the end. The EventType is a series of flags which controls the generation. The rules for this are described in detail at:
+https://cds.cern.ch/record/855452/files/lhcb-2005-034.pdf
````
Linky 🔗

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+Similarly, the document specifies the conventions for the "NickName" - which also has to be the filename. Note that once MC has been produced from a given DecFile, it is not allowed to be changed, so you never need to worry about which version of DecFiles you are looking at when trying to understand existing samples.
+
+The "Cuts" field specifies one of a set of C++ selections in:
+https://gitlab.cern.ch/lhcb/Gauss/blob/master/Gen/GenCuts/
```
🔗 maybe:

"The "Cuts" field specifies which 'cut tool' to use out of a predefined set. The [source code](https://gitlab.cern.ch/lhcb/Gauss/blob/master/Gen/GenCuts/) of the tool you're using should be consulted to fid out what it does."

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+    <before> 27 </before>
+    <value> 0.18519 </value>
+    <error> 0.074757 </error>
+</efficiency>
```
Could wrap this in a code block:

```
```xml
<efficiency name = "generator level cut">
    …
</efficiency>
``` 
```

alexpearce commented on simulation.md:
````diff
@@ -0,0 +1,454 @@
+#
+End
+```
+This DecFile defines a Signa B+ which decays 100% to D0 pi+, and the D0 in turn decays 100% into K- pi+. Important is the definition of "MyD0". If the decay was to "D0" rather than "MyD0", the D0 would decay via all of the decay modes implemented in DECAY.DEC.
````
I don't think `DECAY.DEC` has been mentioned yet. I think it's worth saying something about it, perhaps with a link to it.

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+Note that the fractions will always be renormalised to sum to 1 - you can directly use PDG branching fractions without having to rescale by hand.
+
+## Final state radiation
+Note that PHOTOS is enabled by default, even though many decfiles explicitly specify it. It needs to be explicitly removed via "noPhotos"
```
Worth an extra sentence explaining what PHOTOS does?

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,454 @@
+decay = '[D0 --> K- pi+]CC'
+
+pt_selection = FilterDesktop(
+    'D0_PT_selector', Code="(CHILD(PT, '{0}') > 3000*MeV)".format(decay))
```
You don't need a wrapping `CHILD` functor, because the line selected untagged D0.

alexpearce reviewed (changes requested):
Really good stuff, thanks a lot. I think ideally this should be broken up into 2 or 3 lessons, but I suppose we don't have much time for that.

gciezare committed:
implemented Alex's comments in DecFiles section

gciezare committed:
implemented Alex's comments in DecFiles section

dominikmuller committed:
Alex comments

alexpearce commented on simulation.md:
```diff
@@ -0,0 +1,456 @@
+Dzeros = AutomaticData('/Event/Turbo/'+line+'/Particles')
+
+pt_selection = FilterDesktop(
+    'D0_PT_selector', Code='PT')
```
I'm not sure what a cut of just `PT` will do 😄 I guess you meant `PT > 3*GeV` or something.

dominikmuller replied:
...

dominikmuller committed:
Remaining comments

dominikmuller committed:
Added the cut

dominikmuller commented:
Thanks Alex. We have addressed all comments.
I agree splitting it into multiple lessons would be beneficial but we will have to leave it as it is for now.

alexpearce commented:
Great, thanks a lot! 🍰
